api4/post.go

```diff
@@ -437,6 +437,9 @@ func updatePost(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Updating the file_ids of a post is not a supported operation and will be ignored
+	post.FileIds = nil
```

Comment:
Why is this necessary? It looks like the `UpdatePost(_, false)` should handle that.

Reply:
@crspeller When `safeUpdate == false`, `!false` --> `true`:
```
	if !safeUpdate || !false || true {
		newPost.IsPinned = post.IsPinned
		newPost.HasReactions = post.HasReactions
		newPost.FileIds = post.FileIds
		newPost.Props = post.Props
	}
```
When safeUpdate is true, it takes the values from the post being passed in the argument. When it's true, the original values will be maintained. (We seem to use that case only in test scenarios, it  is confusing to me as well.)

Reply:
Ugg. I hate these flags we sometimes have on functions that are super confusing.

Reply:
Yeah, it's quite confusing. I'll open a ticket to track it, because like you I was quite confused as well. I debugged it in the end to be sure.